src/Microsoft.ML.Transforms/Text/WordTokenizeTransform.cs

```diff
@@ -75,6 +73,8 @@ public abstract class ArgumentsBase : TransformInputBase
                 HelpText = "Comma separated set of term separator(s). Commonly: 'space', 'comma', 'semicolon' or other single character.",
                 ShortName = "sep")]
             public string TermSeparators = "space";
+
+            public char[] CharArrayTermSeparators;
```

Comment:
>CharArrayTermSeparators [](start = 26, length = 23)

This by itself is starting to be good, but we could make it even better. It would be nice if this flaw could be fixed not only in the API, but in entry-points as well. We have a mechanism to do this. Consider what we do here.

[https://github.com/dotnet/machinelearning/blob/3cdd3c8b32705e91dcf46c429ee34196163af6da/src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs#L352-L356](https://github.com/dotnet/machinelearning/blob/3cdd3c8b32705e91dcf46c429ee34196163af6da/src/Microsoft.ML.Data/DataLoadSave/Text/TextLoader.cs#L352-L356)

Note the two close arguments. The difference is, you see, the first has `Visibility = CmdLineOnly` and the second has `Visibility = EntryPointsOnly`. The reason for this distinction is we never quite made out command line parser work very well with arrays. (We probably should, we just haven't.)

Note that if you make this change, you will have to update the manifest, the `CSharpApi.cs`, etc. If you need help figuring out how to do that let me know.

Reply:
Thank you for pointing that out! I have made a change in line with the files you were pointing to.

---
In reply to: [221439001](https://github.com/dotnet/machinelearning/pull/1089#discussion_r221439001) [](ancestors = 221439001)

Reply:
Ooh, I didn't know about this. Looks cool.

---
In reply to: [221773438](https://github.com/dotnet/machinelearning/pull/1089#discussion_r221773438) [](ancestors = 221773438,221439001)